src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/LinkedInversionBreakpointsInference.java

```diff
@@ -0,0 +1,935 @@
+        } else if (intervalThree.contains(intervalFive)) {
+            return new ThreeContainsFive(overlappingPair);
+        } else if (compareIntervals(intervalFive, intervalThree) <= 0) {
+            return new FiveInterleavesThree(overlappingPair);
+        } else {
+            return new ThreeInterleavesFive(overlappingPair);
+        }
+    }
+
+    //==================================================================================================================
+
+    /**
+     * Scenario where re-alignment of short reads back to artificial reference haplotypes
+     * is NOT necessary
+     * to resolve which variants are here.
+     */
+    abstract static class NonDegenerateScenario extends OverlappingPairHandler {
+        private static final long serialVersionUID = 1L;
+
+        NonDegenerateScenario(final OverlappingPair overlappingPair) {
+            super(overlappingPair);
+        }
+    }
+    
+    private static final class FiveInterleavesThree extends NonDegenerateScenario {
+        private static final long serialVersionUID = 1L;
+        FiveInterleavesThree(final OverlappingPair overlappingPair) {
+            super(overlappingPair);
+        }
+
+        // TODO: 5/7/18 refactor the following code and learn from the refactoring made in PR #4602
+        @Override
+        public List<VariantContext> toVariants(final ReferenceMultiSource reference,
+                                               final Map<String, List<SVFastqUtils.FastqRead>> assemblyID2Reads,
+                                               final String pathToFastqsDir) {
+
+            final List<String> contigNames =
+                    Stream.concat(overlappingPair.fivePrimeBreakEnd.makeSureAttributeIsList(GATKSVVCFConstants.CONTIG_NAMES),
+                            overlappingPair.threePrimeBreakEnd.makeSureAttributeIsList(GATKSVVCFConstants.CONTIG_NAMES))
+                            .sorted().distinct().collect(Collectors.toList());
+
+            final String sourceIDs = overlappingPair.fivePrimeBreakEnd.getID().replaceAll("_1$", "")
+                    + VCFConstants.INFO_FIELD_ARRAY_SEPARATOR
+                    + overlappingPair.threePrimeBreakEnd.getID().replaceAll("_1$", "");
+
+
+            final List<VariantContextBuilder> result = new ArrayList<>(3);
+
+
+            byte[] refBase = getRefBase(reference, new SimpleInterval(overlappingPair.chr,
+                    overlappingPair.threeIntervalLeftBoundary, overlappingPair.threeIntervalLeftBoundary));
+            final SimpleInterval invertedInterval = new SimpleInterval(overlappingPair.chr,
+                    overlappingPair.threeIntervalLeftBoundary, overlappingPair.fiveIntervalRightBoundary);
+            final VariantContextBuilder inversionBuilder = makeInversion(invertedInterval, Allele.create(refBase, true));
+
+            if ( overlappingPair.threeIntervalLeftBoundary - overlappingPair.fiveIntervalLeftBoundary - 1 > 49 ) {
```

Comment:
Don't we have the 50 value stored as a constant somewhere? Also, what's the rationale for using 50 bases here?